verification_token/management/commands/clean_verification_tokens.py

```diff
@@ -0,0 +1,22 @@
+from datetime import timedelta
+
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+from django.utils import timezone
+
+from verification_token.models import VerificationToken
+
+
+class Command(BaseCommand):
+
+    def handle(self, **options):
+        inactive_and_expired_tokens = VerificationToken.objects.filter(
+            Q(is_active=False) | Q(expires_at__isnull=False, expires_at__lte=timezone.now())
```

Comment:
Shouldn't you delete also tokens without key? since they are not valid, see `is_valid` property.

Reply:
And strictly speaking, you should compare with `lt` only, again check `is_valid` function. It returns true, if `expires_at` equals current time, therefore, such token is still valid an you are deleting it here. For sure, it is a pretty edge condition, but just to be correct. If for some reason `expires_at` would be just date, the situation would not be marginal anymore.

Reply:
Regarding deleting tokens without a key: Yes, that would make sense, but `key` model attribute is required, nonnullable and unique, so I don't think there will be any such keys.

Reply:
OK, following out conversation off GitHub, and since the database does not allow for the state (`key` is `null=False, blank=False, unique=True` removing `key` presence from `is_valid` check.